mdoc-js/src/main/scala/mdoc/modifiers/ScalajsWorker.scala

```diff
@@ -0,0 +1,106 @@
+package mdoc.modifiers
+
+import org.scalajs.linker.interface._
+import java.net.{URL, URLClassLoader}
+import scala.concurrent.Future
+import java.nio.file.{Path, Paths}
+import scala.concurrent.ExecutionContext
+import scala.language.reflectiveCalls
+
+class ScalajsWorkerState(
+    cachedIRFiles: Seq[IRFile],
+    linker: ClearableLinker
+)
+
+class ScalajsWorker(compilerClasspath: Array[URL], claspath: Array[URL]) {
```

Comment:
The cursed mixture of Scala and Java reflection in this class should be replaced with Java-only, see https://github.com/scala-js/scala-js/blob/main/sbt-plugin/src/main/scala/org/scalajs/sbtplugin/LinkerImpl.scala for reference.

Reply:
[e89e05f](https://github.com/scalameta/mdoc/pull/629/commits/e89e05f1679a6dc34cb8f5def944898297e6d217)